src/ProcessingNodes/processingnode.js

```diff
@@ -100,7 +100,8 @@ class ProcessingNode extends GraphNode {
         for (let inputName of definition.inputs) {
             this._inputTextureUnitMapping.push({
                 name: inputName,
-                textureUnit: gl.TEXTURE0 + this._boundTextureUnits
+                textureUnit: gl.TEXTURE0 + this._boundTextureUnits,
+                textureLocation: gl.getUniformLocation(this._program, inputName)
```

Comment:
So here you cache the location and access that in the render loops instead of calling the getter each time? Nice! There are some good tips on that MDN page.

Did you have a look at this one at all?

`Always have vertex attrib 0 array enabled. If you draw with vertex attrib 0 array disabled, you will force the browser to do complicated emulation when running on desktop OpenGL (e.g. on Mac OSX). This is because in desktop OpenGL, nothing gets drawn if vertex attrib 0 is not array-enabled. You can use bindAttribLocation() to force a vertex attribute to use location 0, and use enableVertexAttribArray() to make it array-enabled.`

I couldn't find any calls to `bindAttribLocation` in VideoContext so that could be worth us trying.